In [5]:
import pandas as pd
import numpy as np
import scipy.stats as stats
from scipy.stats import t


In [6]:
data = pd.read_csv("TempPulz.csv")
moski = data[data.SPOL == 1]
zenske = data[data.SPOL == 2]
#privzamemo da sta temperatura in spol porazdeljena normalno

Ocenite povprečje in standardni odklon za telesno temperaturo posebej pri
moških in posebej pri ženskah.


In [7]:
m_povprecje = moski.TEMPERATURA.mean()
z_povprecje = zenske.TEMPERATURA.mean()

m_std = moski.TEMPERATURA.std(ddof=1) #ddof=1 deli z n-1 da dobimo nepristransko cenilko
z_std = zenske.TEMPERATURA.std(ddof=1) 

 Za povprečji iz prejšnje točke določite 95% intervala zaupanja.

Interval izračunamo po formuli: $$ \Big[ \bar{X} - \frac{\hat{\sigma}}{\sqrt{n}}F^{-1} _{Student(n-1) (1-\frac{\alpha}{2})}, \bar{X} + \frac{\hat{\sigma}}{\sqrt{n}}F^{-1} _{Student(n-1) (1-\frac{\alpha}{2})}\Big]$$

In [8]:
n_m = len(moski)
n_z = len(zenske)



def interval_zaupanja_za_mi(mi, sigma, alpha, n):
    K = sigma / np.sqrt(n) * stats.t.ppf(1-alpha/2, n-1) 
    a = mi - K
    b = mi + K
    return ([a,b])


In [9]:
interval_moski = interval_zaupanja_za_mi(m_povprecje, m_std, 0.05, n_m)

interval_zenske = interval_zaupanja_za_mi(z_povprecje, z_std, 0.05, n_z)

In [10]:
print(interval_moski)
print(interval_zenske)

[97.93147218445705, 98.27775858477375]
[98.20961890918225, 98.57807339851006]


Preizkusite domnevo, da imajo moški in ženske v povprečju enako telesno temperaturo.

Računali bomo pri stopnji tveganja $\alpha = 0.05$ in $\alpha = 0.01 $. <br>
$H_0$: Moški in ženske imajo v povprečju enako telesno temperturo.<br>
$H_1$: Moški in ženske v povprečju nimajo iste temperature. <br>
$H_0$: $\mu_m = \mu_z$ <br>
Testna statistika: $T = \frac{\bar{X}_m - \bar{X}_z + \mu_m - \mu_z}{\sqrt{\frac{\sigma_m^2}{N_m} + \frac{\sigma_z^2}{N_z}}}$  <br>
To izahaja iz vsote dveh normalnih porazdelitev, ki je spet normalna. <br>
Če $H_0$ drži, potem je: $T = \frac{\bar{X}_m - \bar{X}_z }{\sqrt{\frac{\sigma_m^2}{N_m} + \frac{\sigma_z^2}{N_z}}} $

In [13]:
T = (m_povprecje - z_povprecje)/np.sqrt( np.square(m_std) / n_m + np.square(z_std) / n_z  )
print("T je: ",T)

T je:  -2.2854345381654997


Hipotezo, da sta povprečja enaka bomo zavrnili če bo $|T| > F^{-1} _{Student(v)}(1-\alpha/2)$, kjer je $v = \frac{(\sigma_m^2/N_m + \sigma_z^2/N_z )^2}{\frac{(\sigma_m^2/N_m)^2}{N_m - 1} +  \frac{(\sigma_z^2/N_z)^2}{N_z - 1}}$

In [19]:
stevec = np.square(np.square(m_std)/n_m + np.square(z_std)/n_z)
imenovalec = np.square(np.square(m_std)/n_m)/(n_m - 1) + np.square(np.square(z_std)/n_z)/(n_z - 1)
v = stevec/imenovalec
print("v je: " ,v)

v je:  127.5103008303909


Ker je $v$ zelo velik gre Studentova porazdelitev proti normalni. <br>
Pri $\alpha = 0.05$ dobimo $F_{normalna}^{-1}(0.975) = 1.96$ <br>
$|T| = 2.285 > 1.96 $ Torej hipotezo zavrnemo. <br>
    <br>
Pri $\alpha = 0.01$ dobimo $F_{normalna}^{-1}(0.995) = 2.85$ <br>
$|T| = 2.285 < 2.85 $ Torej ni dovolj dokazov, da bi lahko hipotezo zavrnili. <br>